File: others/1600-1700/1695. 删除子数组的最大得分.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <string>
#include <functional>
#include <algorithm>
#include <ranges>
#include <numeric>
#include <unordered_set>
#include <cmath>   
#include <memory> 
#include <map>
#include <queue>
#include <cstring>
#include <array> 
#include <bitset>
#include <stack>
#include <set>
#include <random>

using namespace std;

class Solution {
public:
    int maximumUniqueSubarray(vector<int> &nums) {
        unordered_map<int, int> window;
        int left = 0;
        int ans = 0;
        int sum = 0;
        for (int right = 0; right < nums.size(); right++) {
            window[nums[right]]++;
            sum += nums[right];
            while (window[nums[right]] > 1) {
                window[nums[left]]--;
                sum -= nums[left];
                left++;
            }
            ans = max(ans, sum);
        }
        return ans;
    }
};

File: leetcode/1600-1700/1695. 删除子数组的最大得分.py
from typing import List


class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        window = set()
        left = 0
        s = 0
        ans = 0
        for right, x in enumerate(nums):
            while x in window:
                window.remove(nums[left])
                s -= nums[left]
                left += 1

            s += x
            window.add(x)
            ans = max(ans, s)
        return ans
